### ISSUE TITLE
sonos.getAllGroups returns (a Promise of) an array of a promise of an array

### ISSUE THREAD

**Issue by ludlovian** [bencevans/node-sonos#359]
I would expect sonos.getAllGroups to return a Promise of the group data. Ideally always an array of group data if there is only one group

After the recent commit [fca8ec5](https://github.com/bencevans/node-sonos/commit/fca8ec594f906a71055f1bf70743f2a09dd9f27b) it now returns a Promise of an array of the Promise of the group data (often an array)

The current implementation has:
`const zones = this.zoneGroupTopologyService().AllZoneGroups()`
`return Array.isArray(zones) ? zones : [zones]`

`AllZoneGroups` is an async function, so `zones` is always a Promise, hence it is always wrapped in the array in the next line.

If the line is changed to `await` the zone groups
`const zones = await this.zoneGroupTopologyService().AllZoneGroups()`

Then I think the function will return (a Promise of) an array of group information, even if one group. I suspect this is what was intended.

(Apologies if anything is unlcear or I have missed a process step, this is my first ever Github issue...)

**Node version**: 8.12
**node-sonos version**: 1.5.4
**OS**: Linux (Ubuntu 18.04)

**Comment by svrooij** [comment]
You’re totally right. How about after your first github issue you create a PR to fix it?

I’ll have a look tomorrow
